### ISSUE TITLE
mapFailure unchecked generic array creation

### ISSUE THREAD

**Issue by xenoterracide** [vavr-io/vavr#2647]
```java
  @Override
  default Path getConfigDir() {
    return Try.ofSupplier( this::getClass )
      .map( Class::getClassLoader )
      .map( cl -> cl.getResource( "config" ) )
      .filter( Objects::nonNull )
      .mapTry( URL::toURI )
      .map( Path::of )
      .mapFailure(
        API.Case( API.$( Predicates.instanceOf( URIException.class ) ), Function.identity() ) )
      .get();
  }
```

I'm getting

```
warning: [unchecked] unchecked generic array creation for varargs parameter of type Case<? extends Throwable,? extends Throwable>[]
      .mapFailure(
                 ^
1 warning
```

would be nice not to get that. I see that the code has the `@SuppressWarnings` so I'm not entirely certain what `Java 11` is up to.

```
------------------------------------------------------------
Gradle 6.8.3
------------------------------------------------------------

Build time:   2021-02-22 16:13:28 UTC
Revision:     9e26b4a9ebb910eaa1b8da8ff8575e514bc61c78

Kotlin:       1.4.20
Groovy:       2.5.12
Ant:          Apache Ant(TM) version 1.10.9 compiled on September 27 2020
JVM:          11.0.10 (AdoptOpenJDK 11.0.10+9)
OS:           Linux 5.10.19-1-MANJARO amd64
```

```
io.vavr:vavr:0.+ -> 0.10.3
\--- runtimeClasspath
```

**Comment by danieldietrich** [comment]
I think `Try.mapFailure(...)` needs to be annotated with `@SafeVarargs`. This surpasses warnings at the call site.

> A programmer assertion that the body of the annotated method or constructor does not perform potentially unsafe operations on its varargs parameter. Applying this annotation to a method or constructor suppresses unchecked warnings about a non-reifiable variable arity (vararg) type and suppresses unchecked warnings about parameterized array creation at call sites.

**Comment by pivovarit** [comment]
Now I remember why it was marked as `v1.0.0` - this is an interface method which can't be marked as `@SafeVarargs` because it's abstract. This is a design failure which can be corrected by introducing a breaking change and will be shipped this year as part of `1.0.0`.

I will keep this open to remember about that.
